src/renderer/components/catalog/SearchForm.tsx

```diff
@@ -62,18 +62,20 @@ class Search extends React.Component<IProps, undefined> {
 
         const value = this.inputRef?.current?.value;
         if (!value) {
-            this.props.redirect({
-                pathname: "/library/search/all",
-                search: undefined,
+            this.props.history.push({
+                ...this.props.location,
+                ...{
+                    pathname: "/library/search/all",
+                },
```

Comment:
In my tests, and according to the React Router documentation, the `history` API does not handle `state`, thus why I used the `<Redirect/>` React Component (as a HOC, but we can also use it directly inside each implicated component class).
I will have to run tests again with this PR again to find the edge cases where the state (`displayType`) is not preserved across navigations, for example inside OPDS (search and browsing), inside local bookshelf (search and browsing), as well as when navigating across OPDS/local bookshelf.